File: src/main/kotlin/at/fhhgb/mc/component/sim/sectorizing/GridSectors.java
package at.fhhgb.mc.component.sim.sectorizing;

import at.fhhgb.mc.component.sim.domain.FloorManager;
import at.fhhgb.mc.component.sim.domain.Matrix;
import at.fhhgb.mc.component.sim.model.Position;
import at.fhhgb.mc.component.sim.model.simulator.Floor;
import at.fhhgb.mc.component.sim.view.DrawingContext;

import java.awt.*;

/*
* Name: Michael Nigl
*
* ID: 1510455019
*/
public class GridSectors implements ISectorizing {

    Position position;
    private int xGrid = 3;
    private int yGrid = 3;

    public GridSectors() {

    }

    @Override
    public void createSectors(FloorManager floorManager, Graphics g, Matrix m_tMatrix) {
        Floor floor = floorManager.getFloor();
        if (!(yGrid == 1 && xGrid == 1)) {
            int cellHeight = (int) ((floor.getFloorHeight() * floorManager.getHeightMultiplikator()) / yGrid);
            int cellWidth = (int) ((floor.getFloorWidth() * floorManager.getWidthMultiplikator()) / xGrid);
            for (int y = floorManager.getOffsetTopInPixel(); y < floorManager.getOffsetBottomInPixel() - cellHeight / 2; y += cellHeight) {
                for (int x = floorManager.getOffsetLeftInPixel(); x < floorManager.getOffsetRightInPixel() - cellWidth / 2; x += cellWidth) {
                    Rectangle rectangle = new Rectangle(x, y, cellWidth, cellHeight);
                    if (position != null && rectangle.contains(new Point(position.getX(), position.getY()))) {
                        DrawingContext.drawGrid(rectangle, g, m_tMatrix, true);
                    } else {
                        DrawingContext.drawGrid(rectangle, g, m_tMatrix, false);
                    }
                }
            }
        }
    }

    @Override
    public void addCurrentPosition(Position position) {
        this.position = position;
    }

}

File: src/main/kotlin/at/fhooe/mc/wifipositioning/model/sectoring/GridSectors.kt
package at.fhooe.mc.wifipositioning.model.sectoring

import at.fhooe.mc.wifipositioning.model.graphics.FloorManager
import at.fhooe.mc.wifipositioning.model.graphics.Matrix
import at.fhooe.mc.wifipositioning.model.simulation.Position
import at.fhooe.mc.wifipositioning.model.graphics.DrawingContext

import java.awt.*

class GridSectors : ISectoring {

    internal var position: Position? = null
    private val xGrid = 3
    private val yGrid = 3

    override fun createSectors(floorManager: FloorManager, g: Graphics, m_tMatrix: Matrix) {
        val floor = floorManager.floor
        val cellHeight = (floor!!.floorHeight * floorManager.heightMultiplikator / yGrid).toInt()
        val cellWidth = (floor.floorWidth * floorManager.widthMultiplikator / xGrid).toInt()
        var y = floorManager.offsetTopInPixel

        while (y < floorManager.offsetBottomInPixel - cellHeight / 2) {
            var x = floorManager.offsetLeftInPixel

            while (x < floorManager.offsetRightInPixel - cellWidth / 2) {
                val rectangle = Rectangle(x, y, cellWidth, cellHeight)

                if (position != null && rectangle.contains(Point(position!!.x, position!!.y))) {
                    DrawingContext.drawGrid(rectangle, g, m_tMatrix, true)
                } else {
                    DrawingContext.drawGrid(rectangle, g, m_tMatrix, false)
                }

                x += cellWidth
            }

            y += cellHeight
        }
    }

    override fun addCurrentPosition(position: Position) {
        this.position = position
    }
}
